### PR TITLE
TGUI Latejoin Menu! (#71531)

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request

Converts the latejoin menu to TGUI.

## How Does This Help ***Gameplay***?

Less shitty latejoin menu.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary> 

</details>

## Changelog

:cl: RimiNosha (original), nevimer (porter)
add: TGUI Latejoin Menu
/:cl:

### CODE REVIEW

File: code/modules/mob/dead/new_player/new_player.dm [hunk 1]
```diff
@@ -38,81 +39,6 @@
 /mob/dead/new_player/prepare_huds()
 	return
 
-/mob/dead/new_player/Topic(href, href_list[])
```
Comment (RimiNosha) [inline code/modules/mob/dead/new_player/new_player.dm:41]: So, TG realised after it was merged, but removing this proc wholesale breaks polls. Please restore lines 41-49, and lines 108-114.

File: tgui/packages/tgui/interfaces/common/JobToIcon.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// Used to handle jobs that don't have a trim.
+export const JobToIcon = {
```
Comment (RimiNosha) [inline tgui/packages/tgui/interfaces/common/JobToIcon.ts:2]: Remove this file entirely, and please just port orbit icons, even if they're used for just this menu.

I had a quick glance at the menu without the icons, and it doesn't look nice to me.

See yogstation's port of this to see what you'd need to do. It's surprisingly little.

**Review by RimiNosha — CHANGES_REQUESTED** [review]

**Comment by RimiNosha** [comment]
PRs to also put on top of this before I can accept this:

https://github.com/tgstation/tgstation/pull/72263
https://github.com/tgstation/tgstation/pull/71868
https://github.com/tgstation/tgstation/pull/71883
https://github.com/tgstation/tgstation/pull/71959

File: tgui/packages/tgui/interfaces/common/JobToIcon.ts (hunk 2 above)
Comment (RimiNosha) [reply tgui/packages/tgui/interfaces/common/JobToIcon.ts:2]: I hate this file, but this can be removed for DM-sided everything later.

**Comment by RimiNosha** [comment]
Still missing https://github.com/tgstation/tgstation/pull/72263

File: tgui/packages/tgui/components/index.js [hunk 3]
```diff
@@ -37,6 +37,7 @@ export { RestrictedInput } from './RestrictedInput';
 export { RoundGauge } from './RoundGauge';
 export { Section } from './Section';
 export { Slider } from './Slider';
+export { StyleableSection } from './StyleableSection';
```
Comment (RimiNosha) [inline tgui/packages/tgui/components/index.js:40]: Remove this line.

**Review by RimiNosha — CHANGES_REQUESTED** [review]
Make this change, and it should be good to merge.

**Review by RimiNosha — CHANGES_REQUESTED** [review]
Do this, and it should pass lints.

**Review by RimiNosha — APPROVED** [review]
Oh hey, it passed.

**Comment by nevimer** [comment]
https://user-images.githubusercontent.com/77420409/228135823-2fe8a3a1-55e2-4c4b-8a79-29059a37ded4.mp4

**Pull request merged**